sdk/core/core-client/src/serviceClient.ts

```diff
@@ -29,8 +29,20 @@ export interface ServiceClientOptions extends CommonClientOptions {
   /**
    * If specified, this is the base URI that requests will be made against for this ServiceClient.
    * If it is not specified, then all OperationSpecs must contain a baseUrl property.
+   * @deprecated
    */
   baseUri?: string;
+  /**
+   * If specified, this is the endpoint that requests will be made against for this ServiceClient.
+   * If it is not specified, then all OperationSpecs must contain a baseUrl property.
+   * to encourage customer to use endpoint, we mark the baseUri as deprecated.
```

Comment:
I'd remove this from the docs in endpoint and add. `This property is deprecated and will be removed soon, please use endpoint instead`  in the TSDocs for `baseUri`